File: 2022/day1/solution.go
package main

import (
	// "fmt"
	"io/ioutil"
	"strconv"
	"strings"
)

func main() {
	content, err := ioutil.ReadFile("./input.txt")
	if err != nil {
		panic("help")
	}
	total := 0
	highestTotal := 0
	lines := strings.Split(string(content), "\n")
	for _, line := range lines {
		if line == "" {
			if total > highestTotal {
				highestTotal = total
			}
			total = 0
		}
		value, _ := strconv.Atoi(line)
		total += value

	}
	println(highestTotal)
}

File: 2022/day1/solution.js
const fs = require('fs')

fs.readFile('./input.txt', 'utf8', (err, data) => {
    if (err) {
        console.error(err)
    }
    split = data.split('\n')
    let total = 0
    let highestTotal = 0

    split.forEach((element, index) => {
        if (element === '') {
            //new elf
            if (total > highestTotal) {
                bestElf = index
                highestTotal = total
            }
            total = 0
            return
        }
        total += parseInt(element)
    })
    console.log(highestTotal)
})
